resources/Microsoft.Windows.Setting.Time/Microsoft.Windows.Setting.Time.psm1

```diff
@@ -0,0 +1,235 @@
+    {
+        $currentState = [Time]::New()
+        $currentState.SetTimeZoneAutomatically = [Time]::GetTimeZoneAutoUpdateStatus()
+        $currentState.TimeZone = (Get-TimeZone).Id
+        $currentState.ShowSystemTrayDateTime = [Time]::GetShowSystemTrayDateTimeStatus()
+
+        return $currentState
+    }
+
+    [void] Set()
+    {
+        if ($this.Test())
+        {
+            return
+        }
+
+        $currentState = $this.Get()
+
+        if ($currentState.SetTimeZoneAutomatically -ne $this.SetTimeZoneAutomatically)
+        {
+            Set-ItemProperty -Path $global:tzAutoUpdatePath -Name ([Time]::SetTimeZoneAutomaticallyProperty) -Value $this.SetTimeZoneAutomatically
+        }
+
+        if ($currentState.TimeZone -ne $this.TimeZone)
+        {
+            Set-TimeZone -Id $this.TimeZone    
+        }
+
+        if ($currentState.ShowSystemTrayDateTime -ne $this.ShowSystemTrayDateTime)
+        {
+            $desiredState = switch ($this.ShowSystemTrayDateTime)
+            {
+                $true { "1" } # 1 = Show
+                $false { "0" } # 0 = Hide
+            }
+
+            if ([string]::IsNullOrEmpty((TryGetRegistryValue -Key $global:SysTrayPath -Property ([Time]::ShowSystemTrayDateTimeProperty))))
+            {
+                New-ItemProperty -Path $global:SysTrayPath -Name ([Time]::ShowSystemTrayDateTimeProperty) -Value $desiredState -PropertyType DWORD
+                return
+            }
+
+            Set-ItemProperty -Path $global:SysTrayPath -Name ([Time]::ShowSystemTrayDateTimeProperty) -Value $desiredState
+        }
+    }
+
+    [bool] Test()
+    {
+        $currentState = $this.Get()
+
+        if (($null -ne $this.ShowSystemTrayDateTime) -and ($this.ShowSystemTrayDateTime -ne $currentState.ShowSystemTrayDateTime))
+        {
+            return $false
+        }
+
+        if (($null -ne $this.TimeZone) -and ($this.TimeZone -ne $currentState.TimeZone))
+        {
+            return $false
+        }
+
+        if (($null -ne $this.SetTimeZoneAutomatically) -and ($this.SetTimeZoneAutomatically -ne $currentState.SetTimeZoneAutomatically))
+        {
+            return $false
+        }
+
+        return $true
+    }
+
+    #region Time helper functions
+    static [TimeZoneAutomatically] GetTimeZoneAutoUpdateStatus()
+    {
+        # key should actually always be present, but we'll check anyway
+        $keyValue = TryGetRegistryValue -Key $global:tzAutoUpdatePath -Property ([Time]::SetTimeZoneAutomaticallyProperty)
+        if ($null -eq $keyValue)
+        {
+            return [TimeZoneAutomatically]::NTP
+        }
+        else
+        {
+            return ($keyValue -as [TimeZoneAutomatically])
+        }        
+    }
+
+    static [bool] GetShowSystemTrayDateTimeStatus()
+    {
+        $value = TryGetRegistryValue -Key $global:SysTrayPath -Property ([Time]::ShowSystemTrayDateTimeProperty)
+        if (([string]::IsNullOrEmpty($value)) -or ($null -eq $value))
```

Comment:
You shouldn’t need to check for `($null -eq $value)` since it is already checked in the string method

Reply:
You are right! Modified.